apps/extension/src/service-worker.ts

```diff
@@ -109,17 +111,19 @@ const getServiceHandler = async () => {
       custodyClient ??= createDirectClient(CustodyService, handler, transportOptions);
       stakingClient ??= createDirectClient(StakingService, handler, transportOptions);
 
+      if (new URL(req.url).pathname.startsWith('/penumbra.custody.v1.Custody'))
+        contextValues.set(skCtx, getSpendKey);
```

Comment:
It might be worth considering splitting view-service and custody-service into different packages, because they need very different contexts

Reply:
yes! as context improves this becomes way more possible.